pyroute2/netlink/rtnl/tcmsg/cls_u32.py

```diff
@@ -159,14 +159,20 @@ def cut_field(key, separator):
                 return (new_key, field)
 
             # 'header' array to pack keys to
-            header = [(0, 0) for i in range(256)]
+            header = [(0, 0) for i in range(-256, 256)]
 
             keys = []
             # iterate keys and pack them to the 'header'
             for key in self['keys']:
                 # TODO tags: filter
                 (key, nh) = cut_field(key, '@')  # FIXME: do not ignore nh
-                (key, offset) = cut_field(key, '+')
+                if key.find('-') != -1:
+                  (key, offset) = cut_field(key, '-')
+                  use_minus = True
+                else:
+                  (key, offset) = cut_field(key, '+')
+                  use_minus = False
+
                 offset = int(offset, 0)
                 # a little trick: if you provide /00ff+8, that
                 # really means /ff+9, so we should take it into
```

Comment:
@Vic063 you will have to change the logic of updating offset at line number 185 .i.e.

offset = offset + 1 if not use_minus else offset - 1